pkg/api/customization/podsecuritypolicytemplate/validator.go

```diff
@@ -0,0 +1,27 @@
+package podsecuritypolicytemplate
+
+import (
+	"errors"
+	"github.com/rancher/norman/httperror"
+	"github.com/rancher/norman/types"
+	"github.com/rancher/norman/types/convert"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/kubernetes/pkg/apis/policy"
+	"k8s.io/kubernetes/pkg/apis/policy/validation"
+)
+
+// Validator uses k8s Pod Security Policy validation to prevent creation of pod security policy templates that are invalid
+func Validator(request *types.APIContext, schema *types.Schema, data map[string]interface{}) error {
+	var spec policy.PodSecurityPolicySpec // k8s psp
+	if err := convert.ToObj(data, &spec); err != nil {
+		return httperror.WrapAPIError(err, httperror.InvalidBodyContent, "Pod Security Policy spec conversion error")
+	}
+	var fldPath field.Path
+	var allErrs field.ErrorList
+	allErrs = validation.ValidatePodSecurityPolicySpec(&spec, &fldPath)
+	if len(allErrs) > 0 { // returns the first one, but there may be more
```

Comment:
Rather than making the user correct 1 error at a time, is there any way we can return multiple validation errors to the client?